### PR TITLE
Set charset simple query supported in proxy

### PR DESCRIPTION
**Pull request by soulasuna**
## Set charset simple query supported in proxy

Related to #13317.

Changes proposed in this pull request:
- Set charset simple query supported for PostgreSQL by `SET CLIENT_ENCODING TO 'charsetName';`.
- Set charset simple query supported for MySQL by `SET NAMES 'charsetName';`.

### CODE REVIEW

File: shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-mysql/src/main/java/org/apache/shardingsphere/proxy/frontend/mysql/command/query/text/query/MySQLComQueryPacketExecutor.java [hunk 1]
```diff
@@ -71,10 +80,19 @@ public MySQLComQueryPacketExecutor(final MySQLComQueryPacket packet, final Conne
...
     private Collection<DatabasePacket<?>> processUpdate(final UpdateResponseHeader updateResponseHeader) {
-        responseType = ResponseType.UPDATE;
         return ResponsePacketBuilder.buildUpdateResponsePackets(updateResponseHeader);
     }
     
+    private Collection<DatabasePacket<?>> processClientEncoding(final ClientEncodingResponseHeader clientEncodingResponseHeader) {
+        String value = clientEncodingResponseHeader.getValue();
+        Optional<Charset> charset = MySQLCharacterSet.findByValue(value);
+        if (charset.isPresent()) {
+            clientEncodingResponseHeader.getConnectionSession().getAttributeMap().attr(CommonConstants.CHARSET_ATTRIBUTE_KEY).set(charset.get());
```
Comment (TeslaCN) [inline shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-mysql/src/main/java/org/apache/shardingsphere/proxy/frontend/mysql/command/query/text/query/MySQLComQueryPacketExecutor.java:90]: There is another value `MySQLConstants.MYSQL_CHARACTER_SET_ATTRIBUTE_KEY` which represented MySQL charset and collation is stored in channel attributes map. What if the charset changed to `GBK` while the collation are still `utf8mb4`? The wrong collation id may be returned to client.

**Review by TeslaCN — CHANGES_REQUESTED** [review]

File: shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-mysql/src/main/java/org/apache/shardingsphere/proxy/frontend/mysql/command/query/text/query/MySQLComQueryPacketExecutor.java (hunk 1 above)
Comment (soulasuna) [reply shardingsphere-proxy/shardingsphere-proxy-frontend/shardingsphere-proxy-frontend-mysql/src/main/java/org/apache/shardingsphere/proxy/frontend/mysql/command/query/text/query/MySQLComQueryPacketExecutor.java:90]: When using MySQL set character encoding not just for interacting with the client-side protocol. 
The storage of data will also changed.
I did test. The performance of proxy is consistent with MySQL.

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/encoding/SetClientEncodingBackendHandler.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+import org.apache.shardingsphere.sql.parser.sql.common.statement.dal.SetStatement;
+
+import java.sql.SQLException;
+import java.util.Iterator;
+
+/**
+ * Set client encoding backend handler.
+ */
+@RequiredArgsConstructor
+public class SetClientEncodingBackendHandler implements TextProtocolBackendHandler {
```
Comment (TeslaCN) [inline shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/encoding/SetClientEncodingBackendHandler.java:35]: Consider refactoring this by `DatabaseAdminExecutor` in future.

**Review by TeslaCN — APPROVED** [review]

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/encoding/SetClientEncodingBackendHandler.java (hunk 2 above)
Comment (soulasuna) [reply shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/encoding/SetClientEncodingBackendHandler.java:35]: OK. Only the Simple Query Protocol is currently supported.
Will be refactored to support the JDBC extension protocol.

**Pull request merged**